rs/config/src/execution_environment.rs

```diff
@@ -343,7 +343,7 @@ impl Default for Config {
                 SUBNET_WASM_CUSTOM_SECTIONS_MEMORY_CAPACITY,
             subnet_memory_reservation: SUBNET_MEMORY_RESERVATION,
             max_canister_memory_size: NumBytes::new(
-                MAX_STABLE_MEMORY_IN_BYTES + MAX_WASM_MEMORY_IN_BYTES,
+                MAX_STABLE_MEMORY_IN_BYTES + MAX_WASM64_MEMORY_IN_BYTES,
```

Comment:
This is probably the most tricky change -- when the `MAX_WASM64_MEMORY_IN_BYTES` increases in comparison to `MAX_WASM_MEMORY_IN_BYTES` basically through this config we tell the replica that Wasm32 canisters could have some memory that actually they won't be able to access. 

The alternative is to have 2 `max_canister_memory_size` (one for w32 and one for w64) and pipe them along to the places they are needed. This can also happen in a follow-up PR.